ml_tools/eolearn/ml_tools/TDigestTask.py

```diff
@@ -0,0 +1,199 @@
+        :type kwargs: keyword arguments
+        """
+        super().__init__(*args, **kwargs)
+
+        # check input and output features
+        if isinstance(in_feature, list):
+            self.in_feature = in_feature
+            if isinstance(out_feature, list):
+                assert len(in_feature) == len(out_feature)
+                self.out_feature = out_feature
+            else:
+                raise RuntimeError(
+                    "TDigestTask: you cannot give a list of input features without specifying the a list of output"
+                    " features of the same length!"
+                )
+        else:
+            self.in_feature = [in_feature]
+            if isinstance(out_feature, list):
+                raise RuntimeError(
+                    "TDigestTask: you cannot get a list of T-Digest representations from a single feature!"
+                )
+            else:
+                self.out_feature = [out_feature]
+
+        # check pixelwise parameter
+        if isinstance(pixelwise, list):
+            if len(pixelwise) == len(self.out_feature):
+                self.pixelwise = pixelwise
+            else:
+                assert len(pixelwise) == len(self.out_feature)
+        else:
+            self.pixelwise = [pixelwise] * len(self.out_feature)
+
+        # set mode
+        self.mode = mode
+
+    def execute(self, eopatch):
+        """
+        Execute method that computes the TDigest of the chosen features.
+
+        :param eopatch: EOPatch which the chosen input feature already exists
+        :type eopatch: EOPatch
+        """
+
+        # standard mode
+        if self.mode is None or self.mode == "standard":
+            for in_feature, out_feature, pixelwise in zip(self.in_feature, self.out_feature, self.pixelwise):
+                shape = np.array(eopatch[in_feature].shape)
+                if pixelwise:
+                    eopatch[out_feature] = np.empty(shape[1:], dtype=object)
+                    if len(shape) == 4:  # add feature is temporal check
+                        for i, j, k in product(range(shape[1]), range(shape[2]), range(shape[3])):
+                            eopatch[out_feature][i, j, k] = td.TDigest()
+                            eopatch[out_feature][i, j, k].batch_update(eopatch[in_feature][:, i, j, k].flatten())
+                    elif len(shape) == 3:  # add feature is timeless check
+                        for i, j, k in product(range(shape[0]), range(shape[1]), range(shape[2])):
+                            eopatch[out_feature][i, j, k] = td.TDigest()
+                            eopatch[out_feature][i, j, k].batch_update(eopatch[in_feature][i, j, k].flatten())
+                    else:
+                        raise RuntimeError(
+                            f"TDigestTask: {self.mode} mode only defined for DATA, DATA_TIMELESS, MASK or MASK_TIMELESS"
+                            " feature types!"
+                        )
```

Comment:
to avoid the need for such runtime errors, i suggest you specify the `allowed_feature_types` in the feature parser in the `__init__`. See [here](https://github.com/sentinel-hub/eo-learn/blob/master/core/eolearn/core/utils/parsing.py#L306) for the type of the field.

In your case it should be something like `parse_features(input_features, allowed_feature_types=[FeatureType.DATA, FeatureType.DATA_TIMELESS, ...])`

If different modes require different feature types, that is something that you can also do in the init (so determine the value of a `allowed_types` variable based on `mode` and then use it in the parser).

Reply:
the users still get an error message, it is just not as specific. I would suggest that suitability of inputs is described in the docs e.g. (i'm making stuff up) `- The 'total' mode computes the total T-Digest representation of the whole feature accumulating all timestamps, bands and pixels. Cannot be computed on timeless data.`